model/project.go

```diff
@@ -1631,6 +1633,115 @@ func (p *Project) GetDisplayTask(variant, name string) *patch.DisplayTask {
 	return nil
 }
 
+func GetVersionsWithOptions(projectId string, opts GetVersionsOptions) ([]Version, error) {
+	match := bson.M{
+		VersionIdentifierKey: projectId,
+	}
+	if opts.ByBuildVariant != "" {
```

Comment:
What's the main use case for this part? This should match all versions as long as the variant is defined in the project